Mathlib/Data/Int/Basic.lean

```diff
@@ -0,0 +1,570 @@
+| (ofNat m), (ofNat n) => by simp [Nat.mul_comm]
+| (ofNat m), -[1+ n]   => by simp [Nat.mul_comm]
+| -[1+ m],   (ofNat n) => by simp [Nat.mul_comm]
+| -[1+ m],   -[1+ n]   => by simp [Nat.mul_comm]
+
+lemma of_nat_mul_neg_of_nat (m : ℕ) : ∀ n, ofNat m * neg_of_nat n = neg_of_nat (m * n)
+| 0        => rfl
+| (succ n) => by simp [neg_of_nat, negOfNat] -- TODO: How to avoid having to simp through renaming definitions to unfold them?
+
+lemma neg_of_nat_mul_of_nat (m n : ℕ) : neg_of_nat m * ofNat n = neg_of_nat (m * n) := by
+    rw [Int.mul_comm]
+    simp [of_nat_mul_neg_of_nat, Nat.mul_comm]
+
+lemma neg_succ_of_nat_mul_neg_of_nat (m : ℕ) :
+  ∀ n, -[1+ m] * neg_of_nat n = ofNat (succ m * n)
+| 0        => rfl
+| (succ n) => by simp [neg_of_nat, negOfNat] -- TODO: How to avoid having to simp through renaming definitions to unfold them?
+
+lemma neg_of_nat_mul_neg_succ_of_nat (m n : ℕ) :
+  neg_of_nat n * -[1+ m] = ofNat (n * succ m) := by
+    simp [Int.mul_comm, neg_succ_of_nat_mul_neg_of_nat, Nat.mul_comm]
+
+attribute [local simp] of_nat_mul_neg_of_nat neg_of_nat_mul_of_nat
+  neg_succ_of_nat_mul_neg_of_nat neg_of_nat_mul_neg_succ_of_nat
+
+protected lemma mul_assoc : ∀ a b c : ℤ, a * b * c = a * (b * c)
+| (ofNat m), (ofNat n), (ofNat k) => by simp [Nat.mul_assoc]
+| (ofNat m), (ofNat n), -[1+ k]   => by simp [Nat.mul_assoc]
+| (ofNat m), -[1+ n],   (ofNat k) => by simp [Nat.mul_assoc]
+| (ofNat m), -[1+ n],   -[1+ k]   => by simp [Nat.mul_assoc]
+| -[1+ m],   (ofNat n), (ofNat k) => by simp [Nat.mul_assoc]
+| -[1+ m],   (ofNat n), -[1+ k]   => by simp [Nat.mul_assoc]
+| -[1+ m],   -[1+ n],   (ofNat k) => by simp [Nat.mul_assoc]
+| -[1+ m],   -[1+ n],   -[1+ k]   => by simp [Nat.mul_assoc]
+
+protected lemma mul_zero : ∀ (a : ℤ), a * 0 = 0
+| (ofNat m) => rfl
+| -[1+ m]   => rfl
+
+protected lemma zero_mul (a : ℤ) : 0 * a = 0 :=
+Int.mul_comm a 0 ▸ Int.mul_zero a
+
+lemma neg_of_nat_eq_sub_nat_nat_zero : ∀ n, neg_of_nat n = sub_nat_nat 0 n
+| 0        => rfl
+| (succ n) => rfl
+
+
+
+lemma of_nat_mul_sub_nat_nat (m n k : ℕ) :
+  ofNat m * sub_nat_nat n k = sub_nat_nat (m * n) (m * k) :=
+by
+  cases m with
+  | zero =>
+    simp [of_nat_zero, Int.zero_mul, Nat.zero_mul]
+  | succ m =>
+    have h := Nat.lt_or_le n k
+    cases h with
+    | inl h =>
+      have h' : (succ m) * n < (succ m) * k := Nat.mul_lt_mul_of_pos_left h (Nat.succ_pos m)
+      rw [sub_nat_nat_of_lt h, sub_nat_nat_of_lt h']
+      simp
+      rw [succ_pred_eq_of_pos (Nat.sub_pos_of_lt h)]
+      rw [← neg_of_nat_of_succ, Nat.mul_sub_left_distrib]
+      rw [← succ_pred_eq_of_pos (Nat.sub_pos_of_lt h')]
+      rfl
+    | inr h =>
+      have h' : (succ m) * k ≤ (succ m) * n := Nat.mul_le_mul_left _ h
+      rw [sub_nat_nat_of_le h, sub_nat_nat_of_le h']
+      simp
+      rw [Nat.mul_sub_left_distrib]
+
+
+
+lemma neg_of_nat_add (m n : ℕ) : neg_of_nat m + neg_of_nat n = neg_of_nat (m + n) := by
+  cases m
+  cases n
+  simp
+  simp [Nat.zero_add]
```

Comment:
Can you present this in such a way that it's clearer which goal is being worked on here? e.g. using indentation? You could even just put m and n after the colon and split on them using the equation compiler